docs/2_4_common_co-simulation.adoc

```diff
@@ -255,19 +263,17 @@ This is only allowed if <<canReturnEarly,`canReturnEarly == fmi3True`>>.
 * `earlyReturnTime` is used to tell the FMU at what time to return early from the current <<fmi3DoStep>> (at the earliest), if the return value of <<earlyReturnRequested>> of <<fmi3CallbackIntermediateUpdate>> is `fmi3True`.
 If the <<earlyReturnTime>> is greater than the last signaled <<intermediateUpdateTime>>, the FMU may integrate up to the time instant <<earlyReturnTime>>.
 
+When providing intermediate inputs to the FMU, it is important that the master provides the same input value for the same variable, at the same <<intermediateUpdateTime>>. In other words, it is required that the calculation of inputs to the FMU be deterministic.
```

Comment:
Is that true also if an event occurs? That would be the same intermediateUpdateTime, but a different superdense time step.

Reply:
> 
> 
> Is that true also if an event occurs? That would be the same intermediateUpdateTime, but a different superdense time step.

Intermediate inputs are available only _within_ a communication interval, not at event times. So we are in step mode, and there is no super-dense time. Agreed?